src/backend/marsha/core/api/video.py

```diff
@@ -300,7 +300,11 @@ def start_live(self, request, pk=None):
         if video.live_state is None:
             return Response({"error": "Call initiate-live before starting a live"}, 400)
 
-        if video.live_state not in [defaults.IDLE, defaults.PAUSED]:
+        if video.live_state not in [
+            defaults.IDLE,
+            defaults.STOPPED,
+            defaults.HARVESTED,
+        ]:
             return Response(
                 {
                     "error": (
```

Comment:
line 317 in this file, the AWS is created only if the `live_info` is `None`. Once `harvested`, the `live_info` is not none, it contains the `started_at` and `stopped_at` info. So I think the AWS will never be created again and it will be impossible to restart a live.

Reply:
After several tests it seems to work as expected.
But it looks like aws calls 3 times update_live_state after harvesting, which leads to 500 for the two last ones.
```
app_1                       | {'logGroupName': '/aws/lambda/dev-nico-marsha-mediapackage', 'requestId': '40347094-bee1-4149-b862-650624358262', 'state': 'harvested', 'extraParameters': {'resolutions': [480, 720, 240]}}
app_1                       | {'input': {'id': '1762785', 'endpoints': ['rtmp://34.243.51.55:1935/c1d35ad9-bdb4-4648-a220-060e7586972e_1651785748-primary', 'rtmp://54.155.63.182:1935/c1d35ad9-bdb4-4648-a220-060e7586972e_1651785748-secondary']}, 'channel': {'id': '6287918', 'arn': 'arn:aws:medialive:eu-west-1:082219553157:channel:6287918'}, 'request_ids': ['b358c749-f418-4531-b9b0-a89db7e81a64', 'e47767ed-5a94-476f-b53d-5c80eee82571', '9df91a89-3764-44ee-b300-f44fde6ad76a', '8d58e23c-d684-4d5b-90aa-95c00a762abe']}
app_1                       | HTTP PATCH /api/videos/c1d35ad9-bdb4-4648-a220-060e7586972e/update-live-state/ 200 [0.16, 172.19.0.8:59448]
app_1                       | {'logGroupName': '/aws/lambda/dev-nico-marsha-mediapackage', 'requestId': '2e313e47-c2d4-4e6f-9ee5-1049e28df9ea', 'state': 'harvested', 'extraParameters': {'resolutions': [480, 720, 240]}}
app_1                       | 'medialive'
app_1                       | Internal Server Error: /api/videos/c1d35ad9-bdb4-4648-a220-060e7586972e/update-live-state/
app_1                       | Traceback (most recent call last):
app_1                       |   File "/usr/local/lib/python3.10/site-packages/asgiref/sync.py", line 463, in thread_handler
app_1                       |     raise exc_info[1]
app_1                       |   File "/usr/local/lib/python3.10/site-packages/django/core/handlers/exception.py", line 42, in inner
app_1                       |     response = await get_response(request)
app_1                       |   File "/usr/local/lib/python3.10/site-packages/django/core/handlers/base.py", line 253, in _get_response_async
app_1                       |     response = await wrapped_callback(
app_1                       |   File "/usr/local/lib/python3.10/site-packages/asgiref/sync.py", line 426, in __call__
app_1                       |     ret = await asyncio.wait_for(future, timeout=None)
app_1                       |   File "/usr/local/lib/python3.10/asyncio/tasks.py", line 408, in wait_for
app_1                       |     return await fut
app_1                       |   File "/usr/local/lib/python3.10/site-packages/asgiref/current_thread_executor.py", line 22, in run
app_1                       |     result = self.fn(*self.args, **self.kwargs)
app_1                       |   File "/usr/local/lib/python3.10/site-packages/asgiref/sync.py", line 467, in thread_handler
app_1                       |     return func(*args, **kwargs)
app_1                       |   File "/usr/local/lib/python3.10/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
app_1                       |     return view_func(*args, **kwargs)
app_1                       |   File "/usr/local/lib/python3.10/site-packages/rest_framework/viewsets.py", line 125, in view
app_1                       |     return self.dispatch(request, *args, **kwargs)
app_1                       |   File "/usr/local/lib/python3.10/site-packages/rest_framework/views.py", line 509, in dispatch
app_1                       |     response = self.handle_exception(exc)
app_1                       |   File "/usr/local/lib/python3.10/site-packages/rest_framework/views.py", line 469, in handle_exception
app_1                       |     self.raise_uncaught_exception(exc)
app_1                       |   File "/usr/local/lib/python3.10/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
app_1                       |     raise exc
app_1                       |   File "/usr/local/lib/python3.10/site-packages/rest_framework/views.py", line 506, in dispatch
app_1                       |     response = handler(request, *args, **kwargs)
app_1                       |   File "/app/src/backend/marsha/core/api/video.py", line 518, in update_live_state
app_1                       |     video.live_info["medialive"].update({"request_ids": request_ids})
app_1                       | KeyError: 'medialive'
app_1                       | HTTP PATCH /api/videos/c1d35ad9-bdb4-4648-a220-060e7586972e/update-live-state/ 500 [0.03, 172.19.0.8:59446]
app_1                       | {'logGroupName': '/aws/lambda/dev-nico-marsha-mediapackage', 'requestId': '2e313e47-c2d4-4e6f-9ee5-1049e28df9ea', 'state': 'harvested', 'extraParameters': {'resolutions': [480, 720, 240]}}
app_1                       | 'medialive'
app_1                       | Internal Server Error: /api/videos/c1d35ad9-bdb4-4648-a220-060e7586972e/update-live-state/
app_1                       | Traceback (most recent call last):
app_1                       |   File "/usr/local/lib/python3.10/site-packages/asgiref/sync.py", line 463, in thread_handler
app_1                       |     raise exc_info[1]
app_1                       |   File "/usr/local/lib/python3.10/site-packages/django/core/handlers/exception.py", line 42, in inner
app_1                       |     response = await get_response(request)
app_1                       |   File "/usr/local/lib/python3.10/site-packages/django/core/handlers/base.py", line 253, in _get_response_async
app_1                       |     response = await wrapped_callback(
app_1                       |   File "/usr/local/lib/python3.10/site-packages/asgiref/sync.py", line 426, in __call__
app_1                       |     ret = await asyncio.wait_for(future, timeout=None)
app_1                       |   File "/usr/local/lib/python3.10/asyncio/tasks.py", line 408, in wait_for
app_1                       |     return await fut
app_1                       |   File "/usr/local/lib/python3.10/site-packages/asgiref/current_thread_executor.py", line 22, in run
app_1                       |     result = self.fn(*self.args, **self.kwargs)
app_1                       |   File "/usr/local/lib/python3.10/site-packages/asgiref/sync.py", line 467, in thread_handler
app_1                       |     return func(*args, **kwargs)
app_1                       |   File "/usr/local/lib/python3.10/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
app_1                       |     return view_func(*args, **kwargs)
app_1                       |   File "/usr/local/lib/python3.10/site-packages/rest_framework/viewsets.py", line 125, in view
app_1                       |     return self.dispatch(request, *args, **kwargs)
app_1                       |   File "/usr/local/lib/python3.10/site-packages/rest_framework/views.py", line 509, in dispatch
app_1                       |     response = self.handle_exception(exc)
app_1                       |   File "/usr/local/lib/python3.10/site-packages/rest_framework/views.py", line 469, in handle_exception
app_1                       |     self.raise_uncaught_exception(exc)
app_1                       |   File "/usr/local/lib/python3.10/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
app_1                       |     raise exc
app_1                       |   File "/usr/local/lib/python3.10/site-packages/rest_framework/views.py", line 506, in dispatch
app_1                       |     response = handler(request, *args, **kwargs)
app_1                       |   File "/app/src/backend/marsha/core/api/video.py", line 518, in update_live_state
app_1                       |     video.live_info["medialive"].update({"request_ids": request_ids})
app_1                       | KeyError: 'medialive'
app_1                       | HTTP PATCH /api/videos/c1d35ad9-bdb4-4648-a220-060e7586972e/update-live-state/ 500 [0.02, 172.19.0.8:59450]```